app/models/tenant.rb

```diff
@@ -220,6 +220,10 @@ def self.root_tenant
     in_my_region.roots.first
   end
 
+  def self.root_tenant_name
+    @root_tenant_name ||= root_tenant.name
```

Comment:
I am thinking, what about we memoize `root_tenant`, i.e.:

```
    def self.root_tenant
      @root_tenant ||= in_my_region.roots.first
    end
```

I think the root_tenant shouldn't change, but its name could. Additionally, you wouldn't need this method (and modify callers).

Perhaps I am missing something?